KillerApp/src/main/java/com/example/killerapp/Manager.java

```diff
@@ -0,0 +1,154 @@
+package com.example.killerapp;
+
+import android.app.Activity;
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+import android.util.Log;
+
+import com.dezen.riccardo.smshandler.ReceivedMessageListener;
+import com.dezen.riccardo.smshandler.SMSManager;
+import com.dezen.riccardo.smshandler.SMSMessage;
+import com.dezen.riccardo.smshandler.SMSPeer;
+
+/***
+ * @author Pardeep
+ * Class
+ */
+
+public class Manager {
+    private static Context currentContext;
+    AlarmManager alarmManager = new AlarmManager();
+    LocationManager locationManager=new LocationManager();
+    SMSManager smsManager;
+    Constants constants=new Constants();
+
+    private static final String MANAGER_TAG = "Manager";
+    private  SendResponseSms sendResponseSms;
+
+    public Manager(Context context){
```

Comment:
Could this class become a singleton? Yes? No? Why? I'm not saying it should, I'm saying it's a good class you can use to train on evaluating pros and cons of the singleton pattern, I'd like to hear your thoughts about this

Reply:
Pro: More instance of Manager could create problem if both used on main activity.
Cons: hard to test
To be honest right now i don't know any othr pro / cons

Reply:
Well it kind of makes sense, because this Manager does not define a network, you won't have different networks because the communication is from one device to the lost one